src/game/AI/ScriptDevAI/scripts/kalimdor/dustwallow_marsh.cpp

```diff
@@ -1000,6 +1000,126 @@ UnitAI* GetAI_mob_invis_firework_helper(Creature* pCreature)
+    {
+        Reset();
+        m_creature->SetFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_IMMUNE_TO_PLAYER);
+    }
+
+    uint32 m_attackTimer;
+    uint32 m_breakTimer;
+    uint32 m_sitTimer;
+    bool m_bisAttacking;
+    bool m_bsitDown, m_bstandUp;
+    Creature* attackableDummy;
+
+    void DoCallForHelp(float) override {}
+
+    void HandleAssistanceCall(Unit*, Unit*) override {}
+
+    void Reset() override
+    {
+        SetRootSelf(true);
+        m_attackTimer = 130 * IN_MILLISECONDS;
+        m_breakTimer = 20 * IN_MILLISECONDS;
+        m_bisAttacking = false;
+        m_bsitDown = false;
+        m_bstandUp = false;
+        attackableDummy = GetClosestCreatureWithEntry(m_creature, NPC_THERAMORE_COMBAT_DUMMY, 2.f);
+    }
+
+    void UpdateAI(const uint32 diff) override
+    {
+        if (attackableDummy && attackableDummy->IsAlive())
+        {
+            if (m_bisAttacking)
+            {
+                if (m_attackTimer <= diff)
+                {
+                    m_bisAttacking = false;
+                    m_attackTimer = urand(120, 145) * IN_MILLISECONDS;
+                    m_creature->CombatStop();
+                    attackableDummy->CombatStop();
+                    m_bsitDown = true;
+                    m_sitTimer = 2 * IN_MILLISECONDS;
+                }
+                else
+                {
+                    DoMeleeAttackIfReady();
+                    m_attackTimer -= diff;
+                }
+            }
+            else
+            {
+                if (m_breakTimer <= diff)
+                {
+                    m_bisAttacking = true;
+                    m_breakTimer = urand(15, 60) * IN_MILLISECONDS;
+                    m_creature->AI()->AttackStart(attackableDummy);
+                }
+                else if (m_breakTimer <= 2 * IN_MILLISECONDS && m_bstandUp){
+                    m_creature->SetStandState(UNIT_STAND_STATE_STAND);
+                    m_bstandUp = false;
+                    m_breakTimer -= diff;
+                }
+                else
+                {
+                    if (m_bsitDown)
+                    {
+                        if (m_sitTimer <= diff)
+                        {
+                            m_creature->SetStandState(UNIT_STAND_STATE_SIT);
+                            m_bsitDown = false;
+                            m_bstandUp = true;
+                        }
+                        else
+                        {
+                            m_sitTimer -= diff;
+                        }
+                    }
+                    m_breakTimer -= diff;
+                }
+            }
+        }
+    }
+};
+
+/*######
+## npc_theramore_combat_dummy
+######*/
+
+struct npc_theramore_combat_dummyAI : public ScriptedAI
+{
+    npc_theramore_combat_dummyAI(Creature* pCreature) : ScriptedAI(pCreature)
+    {
+        m_creature->RemoveFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_IMMUNE_TO_NPC);
```

Comment:
Why are you putting it here, why is it in db in the first place.

Reply:
No idea why it's in the DB, but it's causing the Guard to be unable to attack the dummy, it's just standing in the idle "ready to attack animation" and staring at the dummy.
I didn't want to mess with the DB regarding Unit Flags, because I don't know how or why they got there, so I remove them as I need them for the purposes of my script.